extensions/Worker.Extensions.SignalRService/src/Models/SignalRNegotiationContext.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    /// <summary>
+    /// The context of SignalR client negotiation.
+    /// </summary>
+    public sealed class SignalRNegotiationContext
+    {
+        /// <summary>
+        /// The SignalR endpoints with connection information to select.
+        /// </summary>
+        public SignalREndpointConnectionInfo[] Endpoints { get; set; }
+    }
+    
+    /// <summary>
+    /// Contains metadata of an SignalR endpoint and connection information for a client to connect to the SignalR Service.
+    /// </summary>
+    public sealed class SignalREndpointConnectionInfo
+    {
+        /// <summary>
+        /// The type of an endpoint.
+        /// </summary>
+        public SignalREndpointType EndpointType { get; set; }
```

Comment:
We have the other class (`SignalREndpoint`) which has all this info exception the connection info prop. Should we use that type here? (as a prop of this class)

Reply:
The structure of the type is consistent with the original binding. We flatten the structure intentionally so that customers can do things like `return signalRNegotiationContext.Endpoints.Select(e=>e.Online==true).First().ConnectionInfo` in a function. If we change the structure, we need to do extra JSON deserialization customization which still has the problem that might be changed by customers.